lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs

```diff
@@ -250,23 +300,35 @@ combineDbAndLsqData sp =
         pure $ mkApiPool k lsqDefault db
       where
         lsqDefault = PoolLsqData
-            { nonMyopicMemberRewards = minBound
+            { nonMyopicMemberRewards = freshmenMemberRewards
             , relativeStake = minBound
             , saturation = 0
             }
 
-    bothPresent = zipWithMatched mkApiPool
+    -- To give a chance to freshly registered pools that haven't been part of
+    -- any leader schedule, we assign them the average reward of the top @k@
+    -- pools.
+    freshmenMemberRewards
+        = Quantity
+        $ average
+        $ L.take (fromIntegral $ desiredNumberOfStakePools pp)
+        $ L.sort
+        $ map (Down . getQuantity . nonMyopicMemberRewards)
+        $ Map.elems lsqData
+      where
+        average [] = 0
+        average xs = round $ double (sum xs) / double (length xs)
+
+        double :: Integral a => a -> Double
+        double = fromIntegral
 
     mkApiPool
         :: PoolId
         -> PoolLsqData
         -> PoolDbData
         -> Api.ApiStakePool
-    mkApiPool
-        pid
-        (PoolLsqData prew pstk psat)
-        dbData
-        = Api.ApiStakePool
+    mkApiPool pid (PoolLsqData prew pstk psat) dbData =
+        Api.ApiStakePool
```

Comment:
You could save a line here if you wanted to, without going over the line limit.
```suggestion
    mkApiPool pid (PoolLsqData prew pstk psat) dbData = Api.ApiStakePool
```

Reply:
I do love newlines actually, Elm formatted me here :grimacing: